### PR TITLE
win32: add romdata function & etc

### PR DESCRIPTION
**Pull request by taoenwen**
[1] win32: add romdata function
[2] update msvc project for d_superqix

test: 
[romdata.zip](https://github.com/finalburnneo/FBNeo/files/12397863/romdata.zip)
[1] Move the dat file to the /config/romdata/ directory after unzipping
[2] Copy kof98cb.zip and rename it to kof98cp.zip
[3] Copy oldsplusnr.zip and rename it to oldsplusnr211.zip
[4] Move kof98cb.zip and oldsplusnr211.zip to the /roms/romdata/ directory if necessary.
[5] Select a dat file from the menu Load game form rom data

### CODE REVIEW

File: src/burner/win32/main.cpp [hunk 1]
```diff
@@ -1097,7 +1097,7 @@ int ProcessCmdLine()
 								fp = _tfopen(szDat, _T("r"));
 								if (fp) {	// ips dat exists
 									fclose(fp);
-									memset(szDatList[nList], '\0', MAX_PATH * sizeof(TCHAR));
+									memset(szDatList[nList], T('\0'), MAX_PATH * sizeof(TCHAR));
```
Comment (dinkc64) [inline src/burner/win32/main.cpp:1100]: oops, I think a typo?

Comment (taoenwen) [reply src/burner/win32/main.cpp]: I fixed it.

**Comment by taoenwen** [comment]
The implementation of sdl/sdl2 ports is better left to others, I'm not familiar with them at all!

**Comment by dinkc64** [comment]
is it ok to commit now?

**Comment by taoenwen** [comment]
yes，it’s ok

**Comment by dinkc64** [comment]
I had to revert it, because while linking with gcc, gets millions of errors like this:
(I did rm -rf obj before build / make on clean)

errs

**Comment by taoenwen** [comment]
sorry，It doesn't deserve a place.

**Comment by dinkc64** [comment]
taoenwen: I will try to help find the bug with gcc, hopefully soon

best regards,
- dink

**Comment by taoenwen** [comment]
cmd
image

It's strange, it passes when I compile it with gcc or vc

**Comment by taoenwen** [comment]
image

Found some other problems in the sdl autocompile error message, wcscpy is not defined

**Comment by dinkc64** [comment]
taoenwen, please accept my apology, the mistake was entirely my fault.....
very sorry about this!

best regards,
- dink

**Comment by taoenwen** [comment]
@dinkc64 The sdl error compilation message I checked and processed, I'm very new to platforms other than Windows, and would appreciate any help!
https://github.com/finalburnneo/FBNeo/pull/1551

**Comment by dinkc64** [comment]
taoenwen, afraid I am the same regarding SDL!

**Comment by cjom** [comment]
What are this "romdata" functions purpose? Load roms files and CRC's from external files?

https://www.man7.org/linux/man-pages/man3/wcscpy.3.html says one must "#include <wchar.h>"
But probably the "if UNICODE" approach better, anyway.

**Comment by dinkc64** [comment]
I'm curious also, what's it for?

**Comment by barbudreadmon** [comment]
I think that's a feature to show the correct game hack name when running a hack ?

**Comment by taoenwen** [comment]
@cjom 使用外部 romdata（romdesc）的内容来切换与原始驱动程序相对应的静态 romdesc，从而允许运行未列入列表的游戏（包括黑客）。

**Comment by barbudreadmon** [comment]
Afaik, our codebase might require a major rework for unicode support on non-windows platform, with the usage of certain windows-only macros like `TCHAR` as the main issue.

**Pull request merged**